be/src/exprs/jit/jit_expr.cpp

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "exprs/jit/jit_expr.h"
+
+#include <chrono>
+#include <vector>
+
+#include "column/chunk.h"
+#include "common/compiler_util.h"
+#include "common/status.h"
+#include "exprs/anyval_util.h"
+#include "exprs/expr.h"
+#include "exprs/function_context.h"
+#include "exprs/jit/jit_functions.h"
+#include "llvm/IR/IRBuilder.h"
+
+namespace starrocks {
+
+JITExpr* JITExpr::create(ObjectPool* pool, Expr* expr) {
+    TExprNode node;
+    node.node_type = TExprNodeType::JIT_EXPR;
+    node.opcode = TExprOpcode::JIT;
+    node.is_nullable = expr->is_nullable();
+    node.type = expr->type().to_thrift();
+    node.output_scale = expr->output_scale();
+    node.is_monotonic = expr->is_monotonic();
+    return pool->add(new JITExpr(pool, node, expr));
+}
+
+JITExpr::JITExpr(ObjectPool* pool, const TExprNode& node, Expr* expr) : Expr(node), _pool(pool), _expr(expr) {
+    _expr->get_uncompilable_exprs(_children);
+}
+
+Status JITExpr::prepare(RuntimeState* state, ExprContext* context) {
+    Status status = Status::OK();
+    RETURN_IF_ERROR(Expr::prepare(state, context));
+
+    _mem_pool = std::make_unique<MemPool>();
+
+    FunctionContext::TypeDesc return_type = AnyValUtil::column_type_to_type_desc(type());
+    std::vector<FunctionContext::TypeDesc> args_types;
+    for (Expr* child : _children) {
+        args_types.push_back(AnyValUtil::column_type_to_type_desc(child->type()));
```

Comment:
shouldn't loss length info for string type or decimal type.

Reply:
It looks like length info is preserved in the function `AnyValUtil::column_type_to_type_desc`.